kinematics_base_test/src/test_kinematics_plugin.cpp

```diff
@@ -483,23 +520,121 @@ TEST(IKFastPlugin, getIKMultipleSolutions)
       EXPECT_NEAR(poses[0].orientation.z, new_poses[0].orientation.z, IK_NEAR);
       EXPECT_NEAR(poses[0].orientation.w, new_poses[0].orientation.w, IK_NEAR);
     }
+  }
 
+  ROS_INFO_STREAM("Success Rate: "
+                  << (double)success / kinematics_test.num_ik_multiple_tests_);
+  EXPECT_GT(success, 0.99 * kinematics_test.num_ik_multiple_tests_)
+      << "A total of " << num_ik_solutions << " ik solutions were found out of "
+      << kinematics_test.num_ik_multiple_tests_ << " tests.";
+  ROS_INFO_STREAM(
+      "Elapsed time: " << (ros::WallTime::now() - start_time).toSec());
+}
 
-  }
+TEST(IKFastPlugin, getnearestIKSolution) {
+  rdf_loader::RDFLoader rdf_loader_(ROBOT_DESCRIPTION_PARAM);
+  robot_model::RobotModelPtr kinematic_model;
+  const boost::shared_ptr<srdf::Model> &srdf_model = rdf_loader_.getSRDF();
+  const boost::shared_ptr<urdf::ModelInterface> &urdf_model =
+      rdf_loader_.getURDF();
+  kinematic_model.reset(new robot_model::RobotModel(urdf_model, srdf_model));
+  robot_model::JointModelGroup *joint_model_group =
+      kinematic_model->getJointModelGroup(
+          kinematics_test.kinematics_solver_->getGroupName());
 
+  // Test inverse kinematics
+  std::vector<double> seed, fk_values;
+  std::vector<std::vector<double>> solutions;
+  double timeout = 5.0;
+  kinematics::KinematicsQueryOptions options;
+  kinematics::KinematicsResult result;
 
-  ROS_INFO_STREAM("Success Rate: "<<(double)success/kinematics_test.num_ik_multiple_tests_);
-  EXPECT_GT(success , 0.99 * kinematics_test.num_ik_multiple_tests_)<<"A total of "<<num_ik_solutions <<" ik solutions were found out of "
-      <<kinematics_test.num_ik_multiple_tests_<<" tests.";
-  ROS_INFO_STREAM("Elapsed time: "<< (ros::WallTime::now()-start_time).toSec());
-}
+  // getIK
+  moveit_msgs::MoveItErrorCodes error_code;
+  std::vector<double> solution_getIK;
+  solution_getIK.resize(
+      kinematics_test.kinematics_solver_->getJointNames().size(), 0.0);
 
+  std::vector<std::string> fk_names;
+  fk_names.push_back(kinematics_test.kinematics_solver_->getTipFrame());
+  robot_state::RobotState kinematic_state(kinematic_model);
 
+  double min = -3.14;
+  double max = 3.14;
+  double rnd;
 
+  unsigned int success = 0;
+  unsigned int num_ik_solutions = 0;
+  ros::WallTime start_time = ros::WallTime::now();
+  for (unsigned int i = 0; i < kinematics_test.num_nearest_ik_tests_; ++i) {
+    seed.resize(kinematics_test.kinematics_solver_->getJointNames().size(),
+                0.0);
+    fk_values.resize(kinematics_test.kinematics_solver_->getJointNames().size(),
+                     0.0);
+    kinematic_state.setToRandomPositions(joint_model_group);
+    kinematic_state.copyJointGroupPositions(joint_model_group, fk_values);
+    std::vector<geometry_msgs::Pose> poses;
+    poses.resize(1);
+
+    for (std::size_t i = 0; i < seed.size(); i++) {
+      double rnd = (double)rand() / (double)RAND_MAX;
```

Comment:
Does this generate reproducible results?  I believe you have to call `srand()` with a constant seed to ensure this.  A completely random test causes problems when it fails occasionally, but that failure can't be reproduced.